lib/ansible/modules/system/xfs_quota.py

```diff
@@ -0,0 +1,404 @@
+            bhard=dict(type='str'),
+            bsoft=dict(type='str'),
+            ihard=dict(type='int'),
+            isoft=dict(type='int'),
+            mountpoint=dict(type='str', required=True),
+            name=dict(type='str'),
+            rtbhard=dict(type='str'),
+            rtbsoft=dict(type='str'),
+            state=dict(type='str', default='present', choices=['absent', 'present']),
+            type=dict(type='str', required=True, choices=['group', 'project', 'user'])
+        ),
+        supports_check_mode=True,
+    )
+
+    quota_type = module.params['type']
+    name = module.params['name']
+    mountpoint = module.params['mountpoint']
+    bhard = module.params['bhard']
+    bsoft = module.params['bsoft']
+    ihard = module.params['ihard']
+    isoft = module.params['isoft']
+    rtbhard = module.params['rtbhard']
+    rtbsoft = module.params['rtbsoft']
+    state = module.params['state']
+
+    if bhard is not None:
+        bhard = human_to_bytes(bhard)
+
+    if bsoft is not None:
+        bsoft = human_to_bytes(bsoft)
+
+    if rtbhard is not None:
+        rtbhard = human_to_bytes(rtbhard)
+
+    if rtbsoft is not None:
+        rtbsoft = human_to_bytes(rtbsoft)
+
+    result = dict(
+        changed=False,
+    )
+
+    if not os.path.ismount(mountpoint):
+        module.fail_json(msg="Path '%s' is not a mount point" % mountpoint, **result)
+
+    mp = get_fs_by_mountpoint(mountpoint)
+    if mp is None:
+        module.fail_json(msg="Path '%s' is not a mount point or not located on an xfs file system." % mountpoint, **result)
+
+    if quota_type == 'user':
+        type_arg = '-u'
+        quota_default = 'root'
+        if name is None:
+            name = quota_default
+
+        if 'uquota' not in mp['mntopts'] \
+                and 'usrquota' not in mp['mntopts'] \
+                and 'quota' not in mp['mntopts'] \
+                and 'uqnoenforce' not in mp['mntopts'] \
+                and 'qnoenforce' not in mp['mntopts']:
+            module.fail_json(
+                msg='%s is not mounted with the uquota/usrquota/quota/uqnoenforce/qnoenforce option.'
+                    % mountpoint, **result
+            )
+        try:
+            pwd.getpwnam(name)
+        except KeyError as e:
+            module.fail_json(msg="User '%s' does not exist." % name, **result)
+
+    if quota_type == 'group':
+        type_arg = '-g'
+        quota_default = 'root'
+        if name is None:
+            name = quota_default
+
+        if 'gquota' not in mp['mntopts'] and 'grpquota' not in mp['mntopts'] and 'gqnoenforce' not in mp['mntopts']:
+            module.fail_json(
+                msg='%s is not mounted with the gquota/grpquota/gqnoenforce option. (current options: %s)'
+                    % (mountpoint, mp['mntopts']), **result
+            )
+        try:
+            grp.getgrnam(name)
+        except KeyError as e:
+            module.fail_json(msg='User %s doesn\'t exist.' % name, **result)
```

Comment:
Please do it like this everywhere.
```suggestion
            module.fail_json(msg="User '%s' does not exist." % name, **result)
```